juju/juju
# Refactor apiserver ToolsFinder() and ToolsGetter() to take a newEnviron func

## Description of change

The apiserver ToolsGetter and ToolsFnder helpers were constructing an IAAS Environ directly. This change refactors them to take a newEnviron func so that the relevant instance for the model can be constructed. The change allows a k8s model to ask for agent binary metadata even though the agent binaries are not used directly.

Part of the change involved fixing the situation where EnvironConfigGetter had both a State and a Model. Only a model is used now (and an interface at that). As more methods are moved/copied to Model from State, we can eventually avoid using State other than to wrap mgo connections.

## QA steps

bootstrap an aws controller with --agent-version=2.7.2
deploy postgresql with an ebs volume
(ensures storage registry in state works)
run upgrade-controller --dry-run
(check that 2.7.3 is suggested)
upgrade controller
run upgrade-model --dry-run
(check that 2.7.3 is suggested)
upgrade model
